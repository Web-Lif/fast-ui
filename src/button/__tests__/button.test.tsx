import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react';
import Button from '../index';

const sleep = (time: number) => {
    return new Promise<void>((resolve) => {
        setTimeout(() => {
            resolve();
        }, time);
    });
};

test('Button -> test fast double click', async () => {
    const mockClickEvent = jest.fn();

    const { getByText } = render(
        <Button
            onClick={async () => {
                mockClickEvent();
                await sleep(100);
            }}
        >
            button
        </Button>,
    );

    const button = await waitFor(() => getByText('button'));
    fireEvent.click(button);
    fireEvent.click(button);
    act(() => {
        fireEvent.click(button);
    });
    expect(mockClickEvent.mock.calls.length).toBe(1);
});
