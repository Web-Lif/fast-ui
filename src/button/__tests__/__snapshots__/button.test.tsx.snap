// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <button
    class="ant-btn emotion-0 ant-btn-default"
    type="button"
  >
    <span>
      button
    </span>
  </button>
</div>
`;

exports[`test disabled 1`] = `
<div>
  <button
    class="ant-btn emotion-0 ant-btn-default"
    disabled=""
    type="button"
  >
    <span>
      button
    </span>
  </button>
</div>
`;
